homeassistant/components/hue_ble/config_flow.py

```diff
@@ -0,0 +1,201 @@
+"""Config flow for Hue BLE integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from HueBLE import HueBleLight
+import voluptuous as vol
+
+from homeassistant.components import bluetooth
+from homeassistant.components.bluetooth.api import (
+    async_ble_device_from_address,
+    async_scanner_count,
+)
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_MAC, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import device_registry as dr
+
+from .const import DOMAIN, URL_PAIRING_MODE
+
+_LOGGER = logging.getLogger(__name__)
+
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME): str,
+        vol.Required(CONF_MAC): str,
```

Comment:
Requiring the user to manually input a MAC is not friendly, and it will also not work if our bluetooth scanners have not already found the device.

Instead, allow the user to select from a list of previously discovered matching lights, something like this:
https://github.com/home-assistant/core/blob/95b532df41a7139c8f3b24149ae3f07d3a695fdf/homeassistant/components/dormakaba_dkey/config_flow.py#L47-L92

Reply:
The primary way a user is meant to connect is through bluetooth discovery where it automatically shows up in the UI, I included this backup mechanism since I have no way of knowing if the identifier currently used to detect it is the same across all lights as I only have one model of light to test with, I can remove it if its too unfriendly.

Reply:
OK I see.

Raw input of MAC is too unfriendly IMO, it's better to update the manifest with new filters as the need arises.